Get the pages query params

romellogoodman opened the pull request:
**Please describe your one-line program and how to run it.**

It gets the query params of the page and puts them into an object

--------
**What Programming Language?**

javascript

romellogoodman committed:
Get the pages query params

wzhouwzhou commented on javascript/page_query_params.js:
```diff
@@ -0,0 +1 @@
+const pageQueryParams = () => document.location.search.replace(/(^\?)/,'').split('&').reduce(function(o,n) { n=n.split('='); o[n[0]]=n[1]; return o},{});
```
The reduce function
> function(o,n) { n=n.split('='); o[n[0]]=n[1]; return o}

Is actually three lines 🤔

```js
  function(o,n) {
    n=n.split('=');
    o[n[0]]=n[1];
    return o
  }
```
as per defined in #19 (>(i.e. no using ; instead of a new line))
To make this one line though I wonder if this is possible with chained maps/reduces 🤔

romellogoodman commented:
ill give that a try!

kannagikazuko commented:
I have been able to think of a solution! 
```js
const pageQueryParams = () => document.location.search.replace(/(^\?)/, '').split('&').reduce((o, n) => (n => o[n[0]] = n[1])(n.split('=')) && o, {}) || {};
``` 
This solution also returns an empty object if there are no query parameters.

For example,
```js
'?email=someone@example.com'.replace(/(^\?)/, '').split('&').reduce((o,n) => (n => o[n[0]] = n[1])(n.split('=')) && o, {}) || {};
```

produces `{ email: 'someone@example.com' }`

wzhouwzhou commented:
@romellogoodman Would you like to use @kannagikazuko's version or have you thought of another?

romellogoodman committed:
Implementing the solution found https://github.com/kannagikazuko/one-line-wonders/pull/22#issuecomment-429688085

romellogoodman commented:
Implemented the code that @kannagikazuko provided

wzhouwzhou commented:
👍

wzhouwzhou reviewed (commented):
👍  (had to get rid of a stuck review indicator)
